src/rgw/rgw_admin.cc

```diff
@@ -10557,6 +10562,30 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT::PUBSUB_TOPIC_STATS  ) {
+    if (topic_name.empty()) {
+      cerr << "ERROR: topic name was not provided (via --topic)" << std::endl;
+      return EINVAL;
+    }
+
+    std::size_t number_of_reservations;
+    uint32_t number_of_entries;
+    uint64_t queue_size;
+    ret = rgw::notify::get_persistent_queue_by_topic_name(
+        dpp(), static_cast<rgw::sal::RadosStore*>(driver)->getRados()->get_notif_pool_ctx(), topic_name, number_of_reservations,
+        number_of_entries, queue_size, null_yield);
+    if (ret == -ENOENT) {
```

Comment:
i think you can merge the -ENOENT with the other errors. trying to get stats for a topic that is not found is considered an error (unlike deleting a topic which is not found). see: https://github.com/ceph/ceph/pull/52087/files#diff-f3b81fb31115c2616085efcfd6a8daed67a94d1041a03af5de48cc572978f547R10469

Reply:
Done!